src/orion/algo/dehb/dehb.py

```diff
@@ -0,0 +1,509 @@
+            crossover_prob=crossover_prob,
+            mutation_strategy=mutation_strategy,
+            crossover_strategy=crossover_strategy,
+            boundary_fix_type=boundary_fix_type,
+            min_clip=min_clip,
+            max_clip=max_clip,
+            max_age=max_age,
+        )
+
+        # Extract fidelity information
+        fidelity_index = self.fidelity_index
+        if fidelity_index is None:
+            raise RuntimeError(SPACE_ERROR)
+
+        fidelity_dim = space[fidelity_index]
+
+        self.rung = None
+        self.seed = seed
+        self.job_infos = defaultdict(list)
+        self.job_results = dict()
+        self.duplicates = defaultdict(int)
+
+        configspace = self.convert_space(self.space)
+
+        # Initialize
+        self.seed_rng(self.seed)
+        self.dehb = CustomDEHBImpl(
+            self,
+            cs=configspace,
+            configspace=True,
+            dimensions=len(configspace.get_hyperparameters()),
+            mutation_factor=mutation_factor,
+            crossover_prob=crossover_prob,
+            strategy=f"{mutation_strategy}_{crossover_strategy}",
+            min_clip=min_clip,
+            max_clip=max_clip,
+            boundary_fix_type=boundary_fix_type,
+            max_age=max_age,
+            # Derived
+            min_budget=fidelity_dim.low,
+            max_budget=fidelity_dim.high,
+            eta=fidelity_dim.base,
+            # Disable their Dask Integration
+            n_workers=1,
+            client=None,
+            # No need for the user function
+            f=None,
+        )
+        self.rung = len(self.budgets)
+
+    @property
+    def state_dict(self) -> dict:
+        """Return a state dict that can be used to reset the state of the algorithm."""
+        state_dict = super(DEHB, self).state_dict
+        state_dict["DEHB_statedict"] = self.dehb.state_dict
+
+        return deepcopy(state_dict)
+
+    def set_state(self, state_dict: dict) -> None:
+        """Reset the state of the algorithm based on the given state_dict
+
+        :param state_dict: Dictionary representing state of an algorithm
+        """
+        BaseAlgorithm.set_state(self, state_dict)
+        self.dehb.set_state(state_dict["DEHB_statedict"])
+
+    def seed_rng(self, seed: int) -> None:
+        """Seed the state of the random number generator.
+
+        Parameters
+        ----------
+        seed: int
+            Integer seed for the random number generator.
+
+        """
+        self.dehb.seed(seed)
+
+    @property
+    def is_done(self) -> bool:
+        """Return True, if an algorithm holds that there can be no further improvement."""
+        return self.dehb._is_run_budget_exhausted(None, self.rung, None)
+
+    def sample_to_trial(self, sample: np.array, fidelity: int) -> Trial:
+        """Convert a ConfigSpace sample into a trial"""
+        config = self.vector_to_configspace(sample)
+        hps = OrderedDict()
```

Comment:
No need since python 3.7. Just use `dict` instead:
```suggestion
        hps = {}
```